datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GraphQLEngine.java

```diff
@@ -33,18 +34,20 @@
  *
  * <p>In addition, it provides a simplified 'execute' API that accepts a 1) query string and 2) set of variables.
  */
-@Data
+@Getter
 public class GraphQLEngine {
 
     private final GraphQL _engine;
     private final Map<String, Supplier<DataLoader<?, ?>>> _dataLoaderSuppliers;
+    private final DataLoaderRegistry _registry;
 
     private GraphQLEngine(@Nonnull final List<String> schemas,
                           @Nonnull final RuntimeWiring runtimeWiring,
-                          @Nonnull final Map<String, Supplier<DataLoader<?, ?>>> dataLoaderSuppliers) {
+                          @Nonnull final Map<String, Supplier<DataLoader<?, ?>>> dataLoaderSuppliers,
+                            @NonNull final DataLoaderRegistry registry) {
 
         _dataLoaderSuppliers = dataLoaderSuppliers;
-
+        _registry = registry;
```

Comment:
Why are we adding this new private variable?

The registry should be re-created on each request. From the `graphql-java` official documentation https://www.graphql-java.com/documentation/v15/batching/ ( see "Per Request Data Loaders" section ) 

>         //
        // DataLoaderRegistry is a place to register all data loaders in that needs to be dispatched together
        // in this case there is 1 but you can have many.
        //
        // Also note that the data loaders are created per execution request
        //
        DataLoaderRegistry registry = new DataLoaderRegistry();
        registry.register("character", characterDataLoader);

        ExecutionInput executionInput = newExecutionInput()
                .query(getQuery())
                .dataLoaderRegistry(registry)
                .build();



For that reason, I don't include it on creation of the `GraphQLEngine`, but instead recreate for each request using the `dataLoaderSuppliers`.